_posts/2023-11-09-in-progress-wildfly-feature-on-openshift.adoc

```diff
@@ -0,0 +1,310 @@
+
+. Navigate to the *Client Scopes* tab for *myclient*. Here you will see that some scope values are under *Assigned Default Client Scopes* field and some others are under *Assigned Optional Client Scopes* field. When client scopes are added to *Default*, the claims associated with them are automatically added to the access token you receive without the need to configure them. Changing it to *Optional* ensures that they are not included by default and you will only get access to claims if you configure the corresponding scope values. This list of scope values you see here are the values that RH-SSO allows and it varies for different OpenID providers. RH-SSO also allows you to create new scopes, but we will not be doing that for this blog. 
+
+. Now, click on *Roles* and create two roles, *user* and *admin*.
+
+. Finally, create a user called *alice* whose *First Name* is *Alice* and *Last Name* is *Smith* and assign her the *user* and *admin* roles. 
+
+. Next, navigate to the `Credentials` tab and create a password for Alice. Toggle the `Temporary` switch off, so you are not prompted to update the password after the first login.
+
+== Build Container Images for WildFly on OpenShift
+
+Normally to use WildFly features that have been released, we just need to add the WildFly Helm release using the chart that is already available on OpenShift. However, since we are making use of a feature that is not released yet, we will need to build our own image of WildFly from the source code and deploy our application on that. 
+
+=== Clone the WildFly and Elytron Projects
+First we need to clone the `WildFly` and the `WildFly-Elytron` projects with the branches that contain the OIDC scope feature: 
+```
+git clone https://github.com/PrarthonaPaul/wildfly.git
+git checkout WFLY-16532
+```
+Next, go to the directory where the `WildFly-Elytron` project should be cloned and clone it:
+```
+git clone https://github.com/PrarthonaPaul/wildfly-elytron.git
+git checkout WFLY-16532
+```
+Notice that we are not using `WildFly/WildFly` and the `wildfly-security/wildfly-elytron` repositories. Instead, we are
+using a forked versions of the projects. This feature does not add any changes to the `WildFly-Core` project. However,
+to ensure that the branch for `WildFly-Core` is compatible with the branches in the `WildFly` and the `WildFly-Elytron`
+projects, we have created a branch called `WFLY-16532`, which we will use for this guide.
+```
+git clone https://github.com/PrarthonaPaul/wildfly-core.git
+git checkout WFLY-16532
+```
+
+=== Build the WildFly Project
+Next, we need to build these projects to use the local version of the code. To do this, we will specify the SNAPSHOT versions of `Elytron` and `WildFly-core`: 
+```
+cd wildfly-elytron
+mvn clean install -DskipTests
+
+cd ../wildfly-core
+mvn clean install -DskipTests -Dversion.org.wildfly.elytron=<insert_SNAPSHOT_VERSION_OF_ELYTRON> 
```

Comment:
@PrarthonaPaul maybe we had problems before because this should be `-Dversion.org.wildfly.security.elytron` and not `-Dversion.org.wildfly.elytron`

Reply:
good catch. 
Fixed it!